### PR TITLE
A4A: Show a banner based on agency approval status

### PR DESCRIPTION
**Pull request by travisw**
Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/1608
PT: pfunGA-3ie-p2

## Proposed Changes

* We're introducing the ability flag new agencies for review. This change show a banner based on that agency approval status.

Note: There's [an existing issue](https://github.com/Automattic/automattic-for-agencies-dev/issues/1314) for a layout bug when these banners are displayed on mobile screen sizes.

## Why are these changes being made?

* To better inform users about their approval status.

## Testing Instructions

* Also apply this backend PR (169166-ghe-Automattic/wpcom)
* As described in the backend PR, update the blog option for your agency and test the values `pending`, `approved`, and `rejected`.
* Test the following pages to make sure the banner is shown properly (`/overview`; `/marketplace/hosting/wpcom` - including hosting, products, and checkout pages;  `/sites`; `/purchases/payment-methods/add`)

**Pending**
* There should not be a close button shown 
Screenshot 2024-12-20 at 1 48 49 PM

**Rejected**
* There should not be a close button shown 
* The "contact us" link should open the contact form modal
Screenshot 2024-12-20 at 1 49 14 PM

**Approved**
* You might need to hard code `created_at` field, as described in the backend PR if your agency is older than 1 week.
* There should be a close button shown 
* After closing and refreshing the banner should not show again
* On the backend hard code the `created_at` value to be older than 1 week. The banner should no longer be shown in this case
Screenshot 2024-12-20 at 1 48 02 PM

---

Note: You can clear the Calypso preference after closing the banner if need from the UI.

Screenshot 2024-12-20 at 1 43 35 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/components/a4a-agency-approval-notice/index.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+			),
+			level: 'success',
+			hideCloseButton: false,
+		},
+		[ ApprovalStatus.REJECTED ]: {
+			text: translate(
+				'We have not approved your application for the Automattic for Agencies program. Please {{a}}contact support{{/a}} to discuss this further if you think this was done in error.',
+				{
+					components: {
+						a: <a href="#contact-support" />,
```
Comment (jkguidaven) [inline client/a8c-for-agencies/components/a4a-agency-approval-notice/index.tsx]: Can we use the `CONTACT_URL_HASH_FRAGMENT` constant to replace '#contact-support'? This way, if we change the fragment, we only need to update one area.

File: client/a8c-for-agencies/components/a4a-agency-approval-notice/index.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		agency?.approval_status === ApprovalStatus.APPROVED &&
+		agency?.created_at &&
+		new Date( agency.created_at ) < new Date( Date.now() - 7 * 24 * 60 * 60 * 1000 ) // 7 days
+	) {
+		return null;
+	}
+
+	if ( ! agency?.approval_status ) {
+		return null;
+	}
```
Comment (jkguidaven) [inline client/a8c-for-agencies/components/a4a-agency-approval-notice/index.tsx]: I think we can exclude this condition, as the following line covers this already. What do you think?

```
	const bannerDetails = availableBannerDetails[ agency?.approval_status ];

	if ( ! bannerDetails ) {
		return null;
	}
```

**Review by jkguidaven — APPROVED** [review]
Overall, it looks good. I have a few minor comments, but are non-blockers.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17081164

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @travisw for including a screenshot in the description! This is really helpful for our translators.

File: client/a8c-for-agencies/components/a4a-agency-approval-notice/index.tsx (hunk 2 above)
Comment (travisw) [reply client/a8c-for-agencies/components/a4a-agency-approval-notice/index.tsx]: If I just remove that check it gives a linting error: `Type 'undefined' cannot be used as an index type.`
So I've moved it to be part of the `bannerDetails` definition to hopefully make it more concise.

**Comment by travisw** [comment]
Thanks for the feedback @jkguidaven! I've made a couple small updates.

**Comment by travisw** [comment]
I'll merge this now. The banner won't ever show however until the backend PR has also been merged.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17081164).

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17081164).

**Pull request merged**